library/core/src/iter/adapters/mod.rs

```diff
@@ -1,3 +1,4 @@
+// ignore-tidy-filelength
```

Comment:
I can also move all of `Enumerate` into a separate file if that's preferred

Reply:
That's apparently already taken care of in https://github.com/rust-lang/rust/pull/77697